### PR TITLE
fix element trades

### PR DESCRIPTION
**Pull request by jeff-dude**
cc @belen-pruvost @dot2dotseurat @soispoke 

here is a general overview:
- move any `cast()` statement to the base child models which are incremental delta tables, i.e. `element_ethereum.events` or `element_bnb.events` -- this will keep things organized and views simple, should look to apply this to all other protocols feeding `nft.trades/events/fees/mints/burns`
- reformat views to make git / dbt recognize they need rebuilt, since child models are changing datatypes -- until we build a solution to run downstream dependencies, we need to trigger them somehow
- `element_bnb.events` was accidently casting `block_number` as decimal, while all other protocols feeding nft trades has this field as default based on source decoded tables (bigint) -- this caused downstream failures
- reformat `nft.trades` to use our simple formatting from `dex.trades`

### CODE REVIEW

File: models/nft/nft_trades.sql [hunk 1]
```diff
@@ -6,366 +6,57 @@
...
+        amount_original,
+        amount_raw, --team-query cast as decimal: CAST(amount_raw AS DECIMAL(38,0)) AS amount_raw,
+        currency_symbol,
+        currency_contract,
+        nft_contract_address,
+        project_contract_address,
+        aggregator_name,
+        aggregator_address,
+        tx_hash,
+        block_number, --**not sure where** team-query cast as decimal: CAST(number_of_items AS DECIMAL(38,0)) AS number_of_items,
```
Comment (belen-pruvost) [inline models/nft/nft_trades.sql:52]: Not sure if these comments should stay

**Review by belen-pruvost — APPROVED** [review]
This looks good, thank you!

**Review by soispoke — APPROVED** [review]
lgtm, passing all local tests 🤞

File: models/nft/nft_trades.sql (hunk 1 above)
Comment (jeff-dude) [reply models/nft/nft_trades.sql:52]: we should remove those comments in a future PR, that was just my guide and forgotten 🥲

**Pull request merged**